frontend/src/components/APIKeyModal.tsx

```diff
@@ -1,34 +1,37 @@
 import { useState } from 'react';
-
-export default function APIKeyModal({
-  isApiModalOpen,
-  setIsApiModalOpen,
-  apiKey,
+import { useDispatch, useSelector } from 'react-redux';
+import {
   setApiKey,
-}: {
-  isApiModalOpen: boolean;
-  setIsApiModalOpen: React.Dispatch<React.SetStateAction<boolean>>;
-  apiKey: string;
-  setApiKey: React.Dispatch<React.SetStateAction<string>>;
-}) {
+  toggleApiKeyModal,
+  selectIsApiKeyModalOpen,
+} from '../store';
+
+export default function APIKeyModal({}) {

```

Comment:
if the component isnt accepting any props, you could just write it as

`export default function APIKeyModal()`

unlike whats committed, above psuedocode doesnt break the linter.

Reply:
Just forgot to remove curly braces when removing props